test/e2e/cloud_access_role_test.go

```diff
@@ -31,40 +31,33 @@ var _ = Describe("UserLogin", Label("cloud-access-role"), func() {
 	_ = AfterEach(func() {
 		GinkgoWriter.Write([]byte("\n"))
 		GinkgoWriter.Write([]byte("===============================================\n"))
-		GinkgoWriter.Write([]byte("Operator namespace: " + data.Resources.Namespace + "\n"))
+		GinkgoWriter.Write([]byte("Operator namespace: " + testData.Resources.Namespace + "\n"))
 		GinkgoWriter.Write([]byte("===============================================\n"))
 		if CurrentSpecReport().Failed() {
-			SaveDump(&data)
+			SaveDump(testData)
 		}
 		By("Clean Roles", func() {
-			DeleteAllRoles(&data)
+			DeleteAllRoles(testData)
 		})
 		By("Delete Resources, Project with Cloud provider access roles", func() {
-			actions.DeleteUserResourcesProject(&data)
-			actions.DeleteGlobalKeyIfExist(data)
+			actions.DeleteTestDataProject(testData)
+			actions.DeleteGlobalKeyIfExist(*testData)
 		})
 	})
 
 	DescribeTable("Namespaced operators working only with its own namespace with different configuration",
-		func(test model.TestDataProvider, roles []cloudaccess.Role) {
-			data = test
-			cloudAccessRolesFlow(&data, roles)
+		func(test *model.TestDataProvider, roles []cloudaccess.Role) {
+			testData = test
+			actions.ProjectCreationFlow(test)
+			cloudAccessRolesFlow(test, roles)
 		},
 		Entry("Test[cloud-access-role-aws-1]: User has project which was updated with AWS custom role", Label("cloud-access-role-aws-1"),
-			model.NewTestDataProvider(
+			model.SimpleTestDataProvider(
```

Comment:
Not a fan of using the word `simple` here as it implies there is an alternative `DifficultTestDataProvider` or `ToughTestDataProvider`. Instead we still  have `NewTestDataProvider` and newly added `SimpleTestDataProvider`.

It also looks like one of these function can use the other in itself to decrease amount of code repetition 💡